runtime/binding-tls/src/test/java/io/aklivity/zilla/runtime/binding/tls/internal/bench/TlsWorker.java

```diff
@@ -0,0 +1,728 @@
+        final long streamId = frame.streamId();
+
+        final MessageConsumer stream = StreamId.isThrottle(msgTypeId)
+            ? throtllesById.get(streamId)
+            : streamsById.get(streamId);
+
+        stream.accept(msgTypeId, buffer, index, length);
+    }
+
+    private MessageConsumer newStream(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length,
+        MessageConsumer sender)
+    {
+        switch (msgTypeId)
+        {
+        case BeginFW.TYPE_ID:
+            final FrameFW frame = frameRO.wrap(buffer, index, index + length);
+            final long routedId = frame.routedId();
+            final long streamId = frame.streamId();
+
+            if (StreamId.isInitial(streamId))
+            {
+                final BindingHandler handler = handlers.get(routedId);
+                MessageConsumer stream = handler.newStream(msgTypeId, buffer, index, length, this::handleWrite);
+                streamsById.put(streamId, stream);
+                throtllesById.put(streamId, sender);
+
+                long replyId = supplyReplyId(streamId);
+                throtllesById.put(replyId, stream);
+                streamsById.put(replyId, sender);
+            }
+            break;
+        }
+
+        return this::handleWrite;
+    }
+
+    private void handleWrite(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
+    {
+        streamsBuffer.write(msgTypeId, buffer, index, length);
+    }
+
+    private Binding supplyBinding(
+        String type)
+    {
+        return bindings.computeIfAbsent(type, this::createBinding);
+    }
+
+    private Vault supplyVault(
+        String type)
+    {
+        return vaults.computeIfAbsent(type, this::createVault);
+    }
+
+    private Vault createVault(
+        String type)
+    {
+        return vaultFactory.create(type, config);
+    }
+
+    private Binding createBinding(
+        String type)
+    {
+        return factory.create(type, config);
+    }
+
+    private long crc32c(
+        String value)
+    {
+        crc32C.reset();
+        crc32C.update(value.getBytes(UTF_8));
+
+        return crc32C.getValue();
+    }
+
+    private static SignalFW.Builder newSignalRW(
+        int capacity)
+    {
+        MutableDirectBuffer buffer = new UnsafeBuffer(new byte[capacity]);
+        return new SignalFW.Builder().wrap(buffer, 0, buffer.capacity());
+    }
+
+    private final class TlsSignaler implements Signaler
+    {
+        private final ThreadLocal<SignalFW.Builder> signalRW;
+
+        private TlsSignaler()
+        {
+            signalRW = withInitial(() -> newSignalRW(512));
+        }
+
+        @Override
+        public long signalAt(
+            long timeMillis,
+            int signalId,
+            IntConsumer handler)
+        {
+            handler.accept(signalId);
+
+            return System.currentTimeMillis();
```

Comment:
I think we don't need to return current time here, perhaps `NO_CANCEL_ID` instead since these are executed on stack and therefore not cancelable?

Same for other `signalXXX` methods.